corehq/apps/accounting/invoice_pdf.py

```diff
@@ -171,20 +171,21 @@ def get_pdf(self):
             while len(items) > 0:
                 items_to_draw = items[12:]
                 items = items[:12]
-                self.draw_header()
-                self.draw_table(items)
-                self.canvas.showPage()
-                self.canvas.save()
+                if len(items) <= 4:
+                    self.draw_table_and_footer(items)
+                else:
+                    self.draw_header()
+                    self.draw_table(items)
+                    if len(items_to_draw) == 0:
+                        self.canvas.showPage()
+                        self.canvas.save()
+                        self.draw_totals_on_new_page()
```

Comment:
Can just keep this in the if block and leave `showPage` and `save` out?

Reply:
Can also just make `draw_table_and_footer` draw header till footer and let `showPage` and `save` happen here only.

Reply:
I think this can be further improved as
```
    def get_pdf(self):
        if self.is_customer:
            items = self.items
            while len(items) > 0:
                items_to_draw = items[12:]
                items = items[:12]
                # this onwards can be a separate method like `_draw_customer_invoice`
                self.draw_header()
                if len(items) <= 4:
                    self.draw_table_and_footer(items)
                else:
                    self.draw_table(items)
                self.canvas.showPage()
                self.canvas.save()
                if len(items_to_draw) == 0:
                    self.draw_totals_on_new_page()
                items = items_to_draw
        else:
            self.draw_table_and_footer(self.items)

    def draw_table_and_footer(self, items):
        if not self.is_wire or self.is_prepayment:
            self.draw_table(items)
        self.draw_totals(totals_x=inches(5.85), line_height=inches(0.25), subtotal_y=inches(3.5))
        self.draw_footer()
```

Reply:
Ah, that makes much more sense. Thank you!